data-prepper-plugins/kinesis-source/src/test/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisServiceTest.java

```diff
@@ -149,6 +150,7 @@ void setup() {
         kinesisLeaseConfigSupplier = mock(KinesisLeaseConfigSupplier.class);
         kinesisLeaseConfig = mock(KinesisLeaseConfig.class);
         workerIdentifierGenerator = mock(WorkerIdentifierGenerator.class);
+        when(kinesisLeaseConfig.getPipelineIdentifier()).thenReturn(PIPELINE_IDENTIFIER);
```

Comment:
This is changing all the existing tests. Thus, we are not not sufficiently validating this. We should test the condition fully.

I see two options:
1. Make method on `KinesisLeaseConfig` that returns the identifier to use. It can have the conditional that you added in `KinesisService. Then you can test that in isolation. Be sure to add `@JsonIgnore` if you take this approach.
2. Create a new test in here that verifies the behavior with `pipeline_identifier` as well as with no `pipeline_identifer`. The former test should also have a pipeline name that is different so you can be sure the correct name is chosen.

Reply:
I have created separate tests to test against null, empty pipeline identifiers in the `KinesisServiceTest`.